sphinx/parsers.py

```diff
@@ -66,6 +68,26 @@ def get_transforms(self):
         transforms.append(SphinxSmartQuotes)
         return transforms
 
+    def parse(self, inputstring, document):
+        # type: (Any, nodes.document) -> None
+        """Parse text and generate a document tree.
+
+        This derived method accepts StringList as a inputstring parameter.
+        It enables to handle mixed contents (cf. rst_prolog) correctly.
+        """
+        if isinstance(inputstring, StringList):
```

Comment:
This is a new special feature! It allows to parse `StringList` object directly. As a result, we can know where the line comes from (ex. `rst_prolog` and `rst_epilog`).
It will be used on autodoc and i18n modules also.